app/helpers/register_helper.rb

```diff
@@ -4,10 +4,10 @@ def get_description_for_register_field(field)
   end
 
   def registers_by_name
-    OpenRegister.register('register', Rails.configuration.register_phase)
-      ._all_records
-      .select { |item| not_system_register(item) }
-      .map(&:key)
+    register_data = @registers_client.get_register('register', Rails.configuration.register_phase, nil)
+    register_data.get_records
+                 .select { |record| not_system_register(record.entry) }
+                 .sort_by { |record| record.entry.key }
   end
 
   def not_system_register(item)
```

Comment:
Since the `not_system_register` method now takes an entry rather than an item should we rename this parameter to `entry`?